src/System.Runtime.Extensions/src/System/Net/WebUtility.cs

```diff
@@ -330,24 +330,25 @@ public static string UrlEncode(string value)
             if (string.IsNullOrEmpty(value))
                 return value;
 
-            int cSafe = 0;
-            int cSpaces = 0;
+            int unexpandedCount = 0;
+            bool foundSpaces = false;
             for (int i = 0; i < value.Length; i++)
             {
                 char ch = value[i];
                 if (IsUrlSafeChar(ch))
                 {
-                    cSafe++;
+                    unexpandedCount++;
                 }
                 else if (ch == ' ')
                 {
-                    cSpaces++;
+                    unexpandedCount++;
+                    foundSpaces = true;
```

Comment:
@stephentoub Fixed, hadn't thought of it that way; I've remedied this by reverting to the original implementation, and calculating `unexpandedCount` at the end by adding the two counters.